src/GafferScene/Instancer.cpp

```diff
@@ -269,8 +493,116 @@ class Instancer::EngineData : public Data
 			return result;
 		}
 
+		typedef std::map< InternedString, boost::unordered_set< IECore::MurmurHash > > PrototypeHashes;
+
+		// In order to compute the number of variations, we compute a unique hash for every context we use
+		// for evaluating prototypes.  So that we can track which sources are responsible for variations,
+		// we return a vector of hash sets, corresponding to the order of m_prototypeContextVariables,
+		// plus an extra entry at the end for the combined result of all variation sources
```

Comment:
This comment is inaccurate now - we're using a map instead of a vector.

Reply:
Fixed